src/WebSocket.cs

```diff
@@ -102,7 +161,45 @@ public async Task Connect(string? auth, string host, string nameOrAddress, Addre
                             await Ws.CloseAsync(WebSocketCloseStatus.NormalClosure, string.Empty,
                             CancellationToken.None);
                         }
-                        if (OnClose != null) dispatchQueue.Enqueue(() => OnClose(receiveResult.CloseStatus, null));
+                        if (OnClose != null)
+                        {
+                            switch (receiveResult.CloseStatus)
+                            {
+                                case WebSocketCloseStatus.NormalClosure:
+                                    dispatchQueue.Enqueue(() => OnClose(null));
+                                    break;
+                                case WebSocketCloseStatus.EndpointUnavailable:
+                                    dispatchQueue.Enqueue(() => OnClose(new Exception("(1000) The connection has closed after the request was fulfilled.")));
```

Comment:
Is there a reason why the enum is replaced by an exception wrapping a string?

Reply:
Same as above